instrumentation/netty/netty-4-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/common/FutureListenerWrappers.java

```diff
@@ -11,20 +11,25 @@
 import io.netty.util.concurrent.ProgressiveFuture;
 import io.opentelemetry.context.Context;
 import io.opentelemetry.context.Scope;
-import io.opentelemetry.instrumentation.api.caching.Cache;
+import io.opentelemetry.instrumentation.api.cache.Cache;
+import java.lang.ref.WeakReference;
 
 public final class FutureListenerWrappers {
-  // Instead of VirtualField use Cache with weak keys and weak values to store link between original
-  // listener and wrapper. VirtualField works fine when wrapper is stored in a field on original
-  // listener, but when listener class is a lambda instead of field it gets stored in a map with
-  // weak keys where original listener is key and wrapper is value. As wrapper has a strong
-  // reference to original listener this causes a memory leak.
-  // Also note that it's ok if the value is collected prior to the key, since this cache is only
-  // used to remove the wrapped listener from the netty future, and if the value is collected prior
-  // to the key, that means it's no longer used (referenced) by the netty future anyways.
+  // note: it's ok if the value is collected prior to the key, since this cache is only used to
+  // remove the wrapped listener from the netty future, and if the value is collected prior to the
+  // key, that means it's no longer used (referenced) by the netty future anyways.
+  //
+  // also note: this is not using VirtualField in case this is ever converted to library
```

Comment:
Alternatively we could add an option to use weak values when `VirtualField` fallback map is used. When we know that `VirtualField` value references key we could just use `VirtualField` with weak values and would not need to worry whether it is going to use fallback map because something wasn't instrumented or because it is used inside library instrumentation.

Reply:
I'll open a issue for this 👍